library/src/main/java/io/github/boguszpawlowski/composecalendar/month/MonthListState.kt

```diff
@@ -5,33 +5,43 @@ import androidx.compose.runtime.Stable
 import androidx.compose.runtime.derivedStateOf
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.snapshotFlow
-import io.github.boguszpawlowski.composecalendar.header.MonthState
+import io.github.boguszpawlowski.composecalendar.selection.DynamicSelectionState
+import io.github.boguszpawlowski.composecalendar.selection.SelectionState
+import io.github.boguszpawlowski.composecalendar.states.CurrentState
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.flow.launchIn
 import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.launch
+import java.time.LocalDate
 import java.time.YearMonth
 import java.time.temporal.ChronoUnit
 
 @Stable
 internal class MonthListState(
   private val coroutineScope: CoroutineScope,
   private val initialMonth: YearMonth,
-  private val monthState: MonthState,
+  private val currentState: CurrentState,
   private val listState: LazyListState,
+  private val selectionState: DynamicSelectionState?
 ) {
 
   private val currentlyVisibleMonth by derivedStateOf {
     getMonthForPage(listState.firstVisibleItemIndex)
   }
 
   init {
-    snapshotFlow { monthState.currentMonth }.onEach { month ->
+    snapshotFlow { YearMonth.of(
+      currentState.day.year,
+      currentState.day.month
+    ) }.onEach { month ->
```

Comment:
Imo you should only pass the reference to the field you want to observe into `snapshotFlow`. If you want to map it later, you can employ `map` function of regular `Flow`